workflow/validate/validate_test.go

```diff
@@ -1051,6 +1051,105 @@ func TestExitHandler(t *testing.T) {
 	require.NoError(t, err)
 }
 
+var workflowFailuresNotOnExit = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: exit-handlers-
+spec:
+  entrypoint: pass
+  templates:
+  - name: pass
+    container:
+      image: alpine:latest
+      command: [sh, -c]
+      args: ["echo {{workflow.failures}}"]
+`
+
+var workflowFailuresOnExit = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: exit-handlers-
+spec:
+  entrypoint: pass
+  onExit: fail
+  templates:
+  - name: pass
+    container:
+      image: alpine:latest
+      command: [sh, -c]
+      args: ["exit 0"]
+  - name: fail
+    container:
+      image: alpine:latest
+      command: [sh, -c]
+      args: ["echo {{workflow.failures}}"]
+`
+
+var workflowFailuresExitHook = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: exit-handlers-
+spec:
+  entrypoint: pass
+  hooks:
+    exit:
+      template: fail
+  templates:
+  - name: pass
+    container:
+      image: alpine:latest
+      command: [sh, -c]
+      args: ["exit 0"]
+  - name: fail
+    container:
+      image: alpine:latest
+      command: [sh, -c]
+      args: ["echo {{workflow.failures}}"]
+`
+
+var workflowTemplateFailuresExitHook = `
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
```

Comment:
I think you misunderstood https://github.com/argoproj/argo-workflows/pull/12798#discussion_r1530090432. It needs a test against [a template level hook](https://github.com/argoproj/argo-workflows/blob/7a33720398c8152b6da99e006bbe4724ad0bde1c/examples/life-cycle-hooks-tmpl-level.yaml#L11), not an equivalent WorkflowTemplate